service/policy/kasregistry/key_access_server_registry.go

```diff
@@ -212,3 +212,104 @@ func (s KeyAccessServerRegistry) ListKeyAccessServerGrants(ctx context.Context,
 
 	return connect.NewResponse(rsp), nil
 }
+
+func (s KeyAccessServerRegistry) CreateKey(ctx context.Context, r *connect.Request[kasr.CreateKeyRequest]) (*connect.Response[kasr.CreateKeyResponse], error) {
+	s.logger.Debug("creating key", slog.String("keyAccessServer Keys", r.Msg.GetKasId()))
+
+	auditParams := audit.PolicyEventParams{
+		ActionType: audit.ActionTypeCreate,
+		ObjectType: audit.ObjectTypeKasRegistryKeys,
+	}
+
+	resp, err := s.dbClient.CreateKey(ctx, r.Msg)
+	if err != nil {
+		s.logger.Audit.PolicyCRUDFailure(ctx, auditParams)
+		return nil, db.StatusifyError(err, db.ErrTextCreationFailed, slog.String("keyAccessServer Keys", r.Msg.GetKasId()), slog.String("key id", r.Msg.GetKeyId()))
+	}
+
+	auditParams.ObjectID = resp.GetKey().GetKeyId()
+	auditParams.Original = resp.GetKey() // should we be logging this it will have the wrapped KEK and pub key?
```

Comment:
Maybe not, but we should look at other industry tools for logging behaviors and formalize the decision before merging this.